src/cmds/scripts/win_postinstall.py

```diff
@@ -1,633 +1,406 @@
-        print __file__ + ' -u <username> -p <password>'
-        sys.exit(2)
+        largs = ['help', 'user=', 'passwd=']
+        opts, _ = getopt.getopt(sys.argv[1:], 'hu:p:', largs)
+    except getopt.GetoptError as e:
+        print e
+        usage()
+        sys.exit(3)
     for opt, arg in opts:
-        if opt == '-h':
-            print __file__ + ' -u <username> -p <password>'
-            sys.exit()
-        elif opt in ("-u", "--user"):
+        if opt in ('-h', '--help'):
+            usage()
+            sys.exit(0)
+        elif opt in ('-u', '--user'):
             username = arg
-        elif opt in ("-p", "--passwd"):
+        elif opt in ('-p', '--passwd'):
             password = arg
-    ValidateCredentials(username, password)
-    CreatePbsConfFile()
-    InstallingVcredist()
-    InstallDataService(username, password)
-    RunPbsmkdirs()
-    RegistingAndStartingServices(username, password)
-    print "Successfully installed"
-
-
-if __name__ == "__main__":
-    install_path = os.path.dirname(os.path.realpath(__file__))
-    exec_etc_path = os.path.join("exec", "etc")
-    position_of_exec = install_path.find(exec_etc_path)
-    if position_of_exec:
-        pbs_path = install_path[:position_of_exec]
-    else:
-        print "Error while finding the PBS Pro installation path"
+        else:
+            print 'Unreocgnized option %s' % o
+            usage()
+            sys.exit(3)
+    if username is None or password is None:
+        print 'No username and/or password provided!'
+        usage()
+        sys.exit(4)
+
+    if not ctypes.windll.shell32.IsUserAnAdmin():
+        print "Only user with Administrators privileges can run this script!"
+        sys.exit(255)
+    print 'Post installation process started'
+    pbs_conf_path = os.environ.get('PBS_CONF_FILE', None)
+    if pbs_conf_path is None:
+        print "Failed to find PBS_CONF_FILE path from environ"
         sys.exit(1)
-    main(sys.argv[1:])
+    pbs_exec = os.path.join(os.path.dirname(pbs_conf_path), 'exec')
+    if not os.path.isdir(pbs_exec):
+        print "Cound not find PBS_EXEC at %s" % pbs_exec
+        sys.exit(2)
+    pbs_bin = os.path.join(os.path.dirname(pbs_conf_path), 'exec', 'bin')
+    if not os.path.isdir(pbs_bin):
+        print "Cound not find PBS_EXEC\bin at %s" % pbs_bin
+        sys.exit(2)
+    pbs_sbin = os.path.join(os.path.dirname(pbs_conf_path), 'exec', 'sbin')
+    if not os.path.isdir(pbs_bin):
+        print "Cound not find PBS_EXEC\sbin at %s" % pbs_sbin
+        sys.exit(2)
+    pbs_home = os.path.join(os.path.dirname(pbs_conf_path), 'home')
+    validate_cred(username, password)
+    install_vcredist()
+    create_pbs_conf()
+    create_home()
+    init_db(username, password)
+    __run_cmd([os.path.join(pbs_bin, 'pbs_mkdirs.exe')])
+    register_and_start_services(username, password)
+    set_svr_defaults()
+    print 'Successfully completed post installation process'
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
All the print's accompanying the exits seem to be on stdout. They should really be on stderr.

Reply:
@subhasisb ok. will change those to stderr in next commit.